### PR TITLE
Support converting an array of values in Kotlin/Native

### PR DESCRIPTION
**Pull request by JakeWharton**
This is required by Zipline.

### CODE REVIEW

File: zipline/native/Context.cpp [hunk 1]
```diff
@@ -393,7 +393,10 @@ Context::toJavaObject(JNIEnv* env, const JSValueConst& value, bool throwOnUnsupp
 
     case JS_TAG_OBJECT:
       if (JS_IsArray(jsContext, value)) {
-        const auto arrayLength = JS_VALUE_GET_INT(JS_GetPropertyStr(jsContext, value, "length"));
+        auto arrayLengthProperty = JS_GetPropertyStr(jsContext, value, "length");
+        const auto arrayLength = JS_VALUE_GET_INT(arrayLengthProperty);
+        JS_FreeValue(jsContext, arrayLengthProperty);
```
Comment (JakeWharton) [inline zipline/native/Context.cpp:374]: Fixing a leak

Comment (JakeWharton) [reply zipline/native/Context.cpp:374]: Maybe? I'm not clear on which JsValue's need freed

Comment (JakeWharton) [reply zipline/native/Context.cpp:374]: We do it elsewhere for a call to `.length` on a JS array so I'm matching that behavior.

Comment (swankjesse) [reply zipline/native/Context.cpp:374]: Yeah this looks right

Comment (swankjesse) [reply zipline/native/Context.cpp:374]: If we get good at calling close() after all of our tests these leaks will jump out at us

**Review by swankjesse — APPROVED** [review]

**Pull request merged**